File: Array/codeforces_1853A.cpp

// Problem link ===>> https://codeforces.com/problemset/problem/1853/A?locale=en
// submission link ===>> https://codeforces.com/contest/1853/submission/241203154

#include <iostream>
#define INF 10000000000

using namespace std;

long long int t, n, a, minim, b, i, ok;

int main()
{
    cin >> t;
    while (t--)
    {
        cin >> n;
        minim = INF;
        cin >> a;
        for (i = 1, ok = 1; i < n; i++)
        {
            cin >> b;
            if (a > b)
                ok = 0;
            else if ((b - a) / 2 + 1 < minim)
                minim = (b - a) / 2 + 1;
            a = b;
        }
        if (!ok)
            cout << 0 << '\n';
        else
            cout << minim << '\n';
    }
    return 0;
}

File: Array/codeforces_1853A.py

# Problem link ===>> https://codeforces.com/problemset/problem/1853/A?locale=en

def dp(n, k):
  dp = [0] * (k + 1)
  dp[0] = 1
  dp[1] = 1
  for j in range(2, k + 1):
    dp[j] = dp[j - 1] + dp[j - 2]
  return dp[k]

def main():
  t = int(input())
  for _ in range(t):
    n, k = map(int, input().split())
    print(dp(n, k))

if __name__ == "__main__":
  main()